sylvia-derive/src/associated_types.rs

```diff
@@ -47,11 +59,13 @@ impl<'a> AssociatedTypes<'a> {
     }
 
     pub fn as_names(&self) -> Vec<&Ident> {
-        self.0.iter().map(|associated| &associated.ident).collect()
+        self.filtered()
+            .map(|associated| &associated.ident)
+            .collect()
     }
 
-    pub fn as_types_declaration(&self) -> &Vec<&TraitItemType> {
-        &self.0
+    pub fn as_types_declaration(&self) -> Vec<&&TraitItemType> {
```

Comment:
Return an iterator (i.e. it would be the same as `filtered()` method) and let decide the user how to operate on this